Can't start oh-my-posh in Windows using PowerShell

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

### What happened?

Since the 26.0.0 I can't start oh-my-posh in PowerShell, always throw the same error, even without using a config
`C:\Users\Omar: The term 'C:\Users\Omar' is not recognized as a name of a cmdlet, function, script file, or executable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.`

### Theme

Custom config

[config.omp.json](https://github.com/user-attachments/files/20511823/config.omp.json)

### What OS are you seeing the problem on?

Windows

### Which shell are you using?

powershell

### Log output

```Shell
Version: 26.0.2

Shell: oh-my-posh

Prompt:

 Omar Izquierdo   ~                                                                                                                    in oh-my-posh at 16:37:22

Segments:

ConsoleTitle(true)                        -   0 ms
Session(true)                             -   0 ms
Path(true)                                -   1 ms
Git(false)                                -  20 ms
Root(true)                                -   0 ms
Status(true)                              -   0 ms
Node(false)                               -   0 ms
Go(false)                                 -   0 ms
Python(false)                             -   0 ms
Shell(true)                               -   0 ms
Time(true)                                -   0 ms

Run duration: 57.664ms

Cache path: C:\Users\Omar Izquierdo\AppData\Local\oh-my-posh

Config path: no --config set, using default built-in configuration

Logs:

[DEBUG] 16:37:21.961 debug.go:36 → debug mode enabled
[TRACE] 16:37:21.988 load.go:Path() - 0s
[DEBUG] 16:37:21.988 load.go:loadConfig:125 → no config file specified, using default
[TRACE] 16:37:21.988 load.go:loadConfig() - 0s
[TRACE] 16:37:21.988 load.go:Load() - 0s
[DEBUG] 16:37:21.988 terminal.go:Init:54 → plain mode enabled
[TRACE] 16:37:21.988 path.go:Path() - 0s
[DEBUG] 16:37:21.988 file.go:Init:26 → loading cache file: C:\Users\Omar Izquierdo\AppData\Local\oh-my-posh\omp.cache
[DEBUG] 16:37:21.989 file.go:Init:47 → loading cache key: INITVERSIONPWSH
[TRACE] 16:37:21.989 file.go:Init(C:\Users\Omar Izquierdo\AppData\Local\oh-my-posh\omp.cache) - 637.1µs
[TRACE] 16:37:21.989 path.go:Path() - 0s
[DEBUG] 16:37:21.989 file.go:Init:26 → loading cache file: C:\Users\Omar Izquierdo\AppData\Local\oh-my-posh\omp.cache.19488
[ERROR] 16:37:21.989 file.go:Init:32 → open C:\Users\Omar Izquierdo\AppData\Local\oh-my-posh\omp.cache.19488: The system cannot find the file specified.
[TRACE] 16:37:21.989 file.go:Init(C:\Users\Omar Izquierdo\AppData\Local\oh-my-posh\omp.cache.19488) - 0s
[DEBUG] 16:37:21.989 file.go:Get:70 → cache key not found: prompt_count_cache
[TRACE] 16:37:21.989 terminal.go:setPromptCount() - 0s
[TRACE] 16:37:21.989 terminal.go:GOOS() - 0s
[DEBUG] 16:37:21.989 terminal.go:setPwd:115 → C:\Users\Omar Izquierdo
[TRACE] 16:37:21.989 terminal.go:setPwd() - 0s
[TRACE] 16:37:21.989 terminal.go:Init() - 637.1µs
[DEBUG] 16:37:21.989 terminal.go:Shell:368 → no shell name provided in flags, trying to detect it
[DEBUG] 16:37:21.997 terminal.go:Shell:379 → process name: oh-my-posh.exe
[TRACE] 16:37:21.997 terminal.go:Shell() - 7.7039ms
[TRACE] 16:37:21.997 terminal.go:Flags() - 0s
[DEBUG] 16:37:21.997 file.go:Get:70 → cache key not found: template_cache
[TRACE] 16:37:21.997 cache.go:restoreCache() - 0s
[TRACE] 16:37:21.997 terminal_windows.go:Root() - 0s
[TRACE] 16:37:21.997 terminal.go:Shell() - 0s
[TRACE] 16:37:21.997 terminal.go:Flags() - 0s
[TRACE] 16:37:21.997 terminal.go:StatusCodes() - 0s
[TRACE] 16:37:21.997 terminal_windows.go:IsWsl() - 0s
[TRACE] 16:37:21.997 terminal.go:Flags() - 0s
[TRACE] 16:37:21.997 terminal.go:Flags() - 0s
[DEBUG] 16:37:21.997 home.go:12 → C:\Users\Omar Izquierdo
[TRACE] 16:37:21.997 terminal_windows.go:IsWsl() - 0s
[TRACE] 16:37:21.997 terminal.go:Flags() - 0s
[TRACE] 16:37:21.997 terminal.go:Flags() - 0s
[TRACE] 16:37:21.997 terminal.go:GOOS() - 0s
[DEBUG] 16:37:21.997 terminal.go:User:249 → Omar Izquierdo
[TRACE] 16:37:21.997 terminal.go:User() - 0s
[DEBUG] 16:37:21.997 terminal.go:Host:266 → mau5
[TRACE] 16:37:21.997 terminal.go:Host() - 0s
[TRACE] 16:37:21.997 terminal.go:GOOS() - 0s
[DEBUG] 16:37:21.997 terminal.go:Getenv:77 → NO DATA
[TRACE] 16:37:21.997 terminal.go:Getenv(SHLVL) - 0s
[DEBUG] 16:37:21.997 writer.go:Init:106 → terminal program: Windows Terminal
[DEBUG] 16:37:21.997 writer.go:Init:107 → terminal shell: shell
[DEBUG] 16:37:21.997 terminal.go:Getenv:77 → NO DATA
[TRACE] 16:37:21.997 terminal.go:Getenv(OMP_CACHE_DISABLED) - 0s
[DEBUG] 16:37:21.997 file.go:Get:70 → cache key not found: accent_color
[TRACE] 16:37:21.997 terminal_windows.go:WindowsRegistryKeyValue(HKEY_CURRENT_USER\Software\Microsoft\Windows\DWM\ColorizationColor) - 0s
[DEBUG] 16:37:21.997 terminal_windows.go:WindowsRegistryKeyValue:198 → ColorizationColor(DWORD): 0xC400CC6A
[TRACE] 16:37:21.997 colors_windows.go:GetAccentColor() - 0s
[TRACE] 16:37:21.997 colors.go:SetAccentColor() - 0s
[TRACE] 16:37:21.997 terminal.go:Shell() - 0s
[DEBUG] 16:37:21.997 terminal.go:Getenv:77 → NO DATA
[TRACE] 16:37:21.997 terminal.go:Getenv(POSH_SHELL_VERSION) - 0s
[DEBUG] 16:37:21.997 debug.go:PrintDebug:24 → segment: Title
[DEBUG] 16:37:21.997 text.go:patchTemplate:125 → {{ .Shell }} in {{ .Folder }}
[TRACE] 16:37:21.997 text.go:Render({{ .Shell }} in {{ .Folder }}) - 536.2µs
[TRACE] 16:37:21.997 terminal.go:Flags() - 0s
[DEBUG] 16:37:21.997 terminal.go:Getenv:77 → NO DATA
[TRACE] 16:37:21.997 terminal.go:Getenv(POSH_CURSOR_LINE) - 0s
[DEBUG] 16:37:21.997 terminal.go:Getenv:77 → NO DATA
[TRACE] 16:37:21.997 terminal.go:Getenv(POSH_CURSOR_COLUMN) - 0s
[TRACE] 16:37:21.997 terminal.go:Flags() - 0s
[TRACE] 16:37:21.997 terminal.go:Flags() - 0s
[TRACE] 16:37:21.997 terminal.go:Flags() - 0s
[TRACE] 16:37:21.997 terminal.go:Flags() - 0s
[DEBUG] 16:37:21.997 home.go:12 → C:\Users\Omar Izquierdo
[DEBUG] 16:37:21.997 home.go:12 → C:\Users\Omar Izquierdo
[TRACE] 16:37:21.997 terminal.go:GOOS() - 0s
[TRACE] 16:37:21.997 terminal.go:GOOS() - 0s
[DEBUG] 16:37:21.997 segment.go:Execute:112 → segment: Status
[DEBUG] 16:37:21.997 segment.go:Execute:112 → segment: Path
[DEBUG] 16:37:21.997 file.go:Get:70 → cache key not found: toggle_cache
[DEBUG] 16:37:21.997 file.go:Get:70 → cache key not found: toggle_cache
[DEBUG] 16:37:21.997 segment.go:isToggled:229 → no toggles found
[DEBUG] 16:37:21.997 segment.go:isToggled:229 → no toggles found
[TRACE] 16:37:21.997 terminal.go:StatusCodes() - 0s
[DEBUG] 16:37:21.997 properties.go:GetBool:22 → display_cygpath: false
[TRACE] 16:37:21.997 terminal.go:GOOS() - 0s
[TRACE] 16:37:21.997 separator.go:Separator() - 0s
[TRACE] 16:37:21.997 terminal.go:Shell() - 0s
[TRACE] 16:37:21.997 terminal.go:Shell() - 0s
[TRACE] 16:37:21.997 terminal.go:GOOS() - 0s
[TRACE] 16:37:21.997 terminal.go:Flags() - 0s
[DEBUG] 16:37:21.997 home.go:12 → C:\Users\Omar Izquierdo
[TRACE] 16:37:21.997 terminal.go:GOOS() - 0s
[DEBUG] 16:37:21.997 properties.go:GetBool:22 → mapped_locations_enabled: true
[DEBUG] 16:37:21.997 segment.go:Execute:112 → segment: Git
[DEBUG] 16:37:21.997 properties.go:GetString:28 → windows_registry_icon: 
[DEBUG] 16:37:21.997 properties.go:GetString:28 → windows_registry_icon: 
[DEBUG] 16:37:21.997 file.go:Get:70 → cache key not found: toggle_cache
[DEBUG] 16:37:21.997 segment.go:isToggled:229 → no toggles found
[DEBUG] 16:37:21.997 home.go:12 → C:\Users\Omar Izquierdo
[TRACE] 16:37:21.997 separator.go:Separator() - 0s
[TRACE] 16:37:21.997 terminal.go:GOOS() - 0s
[TRACE] 16:37:21.998 terminal.go:GOOS() - 0s
[DEBUG] 16:37:21.998 properties.go:GetString:28 → home_icon: ~
[DEBUG] 16:37:21.998 properties.go:GetKeyValueMap:46 → mapped_locations: map[]
[TRACE] 16:37:21.998 separator.go:Separator() - 0s
[TRACE] 16:37:21.998 terminal.go:GOOS() - 0s
[TRACE] 16:37:21.998 separator.go:Separator() - 0s
[TRACE] 16:37:21.998 terminal.go:GOOS() - 0s
[DEBUG] 16:37:21.998 text.go:patchTemplate:125 → {{ .Code }}
[TRACE] 16:37:21.998 terminal.go:GOOS() - 0s
[TRACE] 16:37:21.998 terminal.go:GOOS() - 0s
[TRACE] 16:37:21.998 terminal.go:GOOS() - 0s
[TRACE] 16:37:21.998 text.go:Render({{ .Code }}) - 538.2µs
[TRACE] 16:37:21.998 terminal.go:Flags() - 0s
[DEBUG] 16:37:21.998 properties.go:GetBool:22 → always_enabled: true
[DEBUG] 16:37:21.998 properties.go:GetStringArray:52 → cycle: []
[TRACE] 16:37:21.998 terminal.go:Flags() - 0s
[DEBUG] 16:37:21.998 properties.go:GetString:28 → folder_separator_template:
[DEBUG] 16:37:21.998 properties.go:GetString:28 → folder_separator_icon: \
[DEBUG] 16:37:21.998 home.go:12 → C:\Users\Omar Izquierdo
[TRACE] 16:37:21.998 terminal.go:GOOS() - 0s
[DEBUG] 16:37:21.998 properties.go:GetBool:22 → cycle_folder_separator: false
[DEBUG] 16:37:21.998 properties.go:GetString:28 → folder_format: %s
[DEBUG] 16:37:21.998 segment.go:Execute:112 → segment: Session
[DEBUG] 16:37:21.998 properties.go:GetString:28 → edge_format: %s
[DEBUG] 16:37:21.998 file.go:Get:70 → cache key not found: toggle_cache
[DEBUG] 16:37:21.998 properties.go:GetString:28 → left_format: %s
[DEBUG] 16:37:21.998 segment.go:isToggled:229 → no toggles found
[DEBUG] 16:37:21.998 properties.go:GetString:28 → right_format: %s
[TRACE] 16:37:21.998 terminal.go:Flags() - 0s
[DEBUG] 16:37:21.998 terminal.go:Getenv:77 → NO DATA
[TRACE] 16:37:21.998 terminal.go:Getenv(SSH_CONNECTION) - 0s
[TRACE] 16:37:21.998 terminal.go:GOOS() - 0s
[DEBUG] 16:37:21.998 terminal.go:Getenv:77 → NO DATA
[TRACE] 16:37:21.998 terminal.go:StackCount() - 0s
[TRACE] 16:37:21.998 terminal.go:Getenv(SSH_CLIENT) - 0s
[TRACE] 16:37:21.998 terminal.go:Flags() - 0s
[DEBUG] 16:37:21.998 text.go:patchTemplate:125 →  {{ if .Data.SSHSession }} {{ end }}{{ .UserName }}
[DEBUG] 16:37:21.998 home.go:12 → C:\Users\Omar Izquierdo
[TRACE] 16:37:21.998 terminal.go:GOOS() - 0s
[DEBUG] 16:37:21.998 segment.go:Execute:112 → segment: Root
[DEBUG] 16:37:21.998 file.go:Get:70 → cache key not found: toggle_cache
[DEBUG] 16:37:21.998 segment.go:isToggled:229 → no toggles found
[TRACE] 16:37:21.998 terminal_windows.go:Root() - 0s
[DEBUG] 16:37:21.999 win32_windows.go:isWriteable:259 → not current user or in group
[DEBUG] 16:37:21.999 win32_windows.go:isWriteable:263 → current user is member of S-1-5-32-544
[DEBUG] 16:37:21.999 win32_windows.go:isWriteable:271 ↓
    WRITE_DAC
    WRITE_OWNER
    SYNCHRONIZE
    DELETE
    READ_CONTROL
[TRACE] 16:37:21.999 text.go:Render( {{ if .SSHSession }} {{ end }}{{ .UserName }} ) - 1.0949ms
[DEBUG] 16:37:21.999 win32_windows.go:isWriteable:273 → user has write access
[TRACE] 16:37:21.999 terminal_windows.go:DirIsWritable() - 1.0949ms
[TRACE] 16:37:21.999 text.go:Render(diamond) - 0s
[DEBUG] 16:37:22.000 text.go:patchTemplate:125 →   {{ path .Data.Path .Data.Location }}
[TRACE] 16:37:22.000 text.go:Render(  {{ path .Path .Location }} ) - 0s
[TRACE] 16:37:22.000 text.go:Render(powerline) - 0s
[DEBUG] 16:37:22.017 terminal.go:CommandPath:317 → C:\Users\Omar Izquierdo\scoop\shims\git.exe
[TRACE] 16:37:22.017 terminal.go:CommandPath(git.exe) - 19.4035ms
[TRACE] 16:37:22.017 terminal.go:HasCommand(git.exe) - 19.4035ms
[DEBUG] 16:37:22.017 properties.go:GetBool:22 → fetch_bare_info: false
[DEBUG] 16:37:22.017 terminal.go:ResolveSymlink:208 → C:\Users\Omar Izquierdo
[TRACE] 16:37:22.017 terminal.go:ResolveSymlink(C:\Users\Omar Izquierdo) - 654.2µs
[ERROR] 16:37:22.017 terminal.go:HasParentFilePath:476 → CreateFile .git: The system cannot find the file specified.
[TRACE] 16:37:22.017 terminal.go:HasParentFilePath(.git) - 654.2µs
[TRACE] 16:37:22.017 text.go:Render(powerline) - 0s
[TRACE] 16:37:22.017 text.go:Render(  ) - 0s
[TRACE] 16:37:22.017 text.go:Render(powerline) - 0s
[DEBUG] 16:37:22.017 text.go:patchTemplate:125 →  {{ if gt .Code 0 }}{{ else }}{{ end }}
[TRACE] 16:37:22.018 text.go:Render( {{ if gt .Code 0 }}{{ else }}{{ end }} ) - 545.6µs
[DEBUG] 16:37:22.018 text.go:patchTemplate:125 → {{ if gt .Code 0 }}p:red{{ end }}
[TRACE] 16:37:22.018 text.go:Render({{ if gt .Code 0 }}p:red{{ end }}) - 0s
[TRACE] 16:37:22.018 text.go:Render(diamond) - 0s
[DEBUG] 16:37:22.018 text.go:patchTemplate:125 → {{ if gt .Code 0 }}p:red{{ end }}
[TRACE] 16:37:22.018 text.go:Render({{ if gt .Code 0 }}p:red{{ end }}) - 0s
[TRACE] 16:37:22.018 terminal.go:Shell() - 0s
[TRACE] 16:37:22.018 terminal.go:Shell() - 0s
[TRACE] 16:37:22.018 terminal.go:Flags() - 0s
[DEBUG] 16:37:22.018 home.go:12 → C:\Users\Omar Izquierdo
[TRACE] 16:37:22.018 terminal.go:GOOS() - 0s
[TRACE] 16:37:22.018 terminal.go:Flags() - 0s
[TRACE] 16:37:22.018 terminal.go:Flags() - 0s
[DEBUG] 16:37:22.018 segment.go:Execute:112 → segment: Time
[DEBUG] 16:37:22.018 file.go:Get:70 → cache key not found: toggle_cache
[DEBUG] 16:37:22.018 segment.go:isToggled:229 → no toggles found
[DEBUG] 16:37:22.018 home.go:12 → C:\Users\Omar Izquierdo
[TRACE] 16:37:22.018 terminal.go:Flags() - 0s
[TRACE] 16:37:22.018 terminal.go:GOOS() - 0s
[DEBUG] 16:37:22.018 segment.go:Execute:112 → segment: Python
[DEBUG] 16:37:22.018 file.go:Get:70 → cache key not found: toggle_cache
[DEBUG] 16:37:22.018 segment.go:isToggled:229 → no toggles found
[TRACE] 16:37:22.018 terminal.go:Flags() - 0s
[DEBUG] 16:37:22.018 properties.go:GetString:28 → time_format: 15:04:05
[DEBUG] 16:37:22.018 home.go:12 → C:\Users\Omar Izquierdo
[DEBUG] 16:37:22.018 home.go:12 → C:\Users\Omar Izquierdo
[DEBUG] 16:37:22.018 segment.go:Execute:112 → segment: Shell
[DEBUG] 16:37:22.018 file.go:Get:70 → cache key not found: toggle_cache
[TRACE] 16:37:22.018 terminal.go:GOOS() - 0s
[DEBUG] 16:37:22.018 file.go:Get:70 → cache key not found: toggle_cache
[DEBUG] 16:37:22.018 segment.go:isToggled:229 → no toggles found
[DEBUG] 16:37:22.018 segment.go:isToggled:229 → no toggles found
[DEBUG] 16:37:22.018 file.go:Get:70 → cache key not found: toggle_cache
[DEBUG] 16:37:22.018 segment.go:isToggled:229 → no toggles found
[DEBUG] 16:37:22.018 properties.go:GetKeyValueMap:46 → mapped_shell_names: map[]
[TRACE] 16:37:22.018 terminal.go:Shell() - 0s
[TRACE] 16:37:22.018 terminal.go:Flags() - 0s
[DEBUG] 16:37:22.018 properties.go:GetStringArray:52 → extensions: [*.js *.ts package.json .nvmrc pnpm-workspace.yaml .pnpmfile.cjs .vue]
[DEBUG] 16:37:22.018 properties.go:GetStringArray:52 → extensions: [*.py *.ipynb pyproject.toml venv.bak]
[DEBUG] 16:37:22.018 properties.go:GetStringArray:52 → extensions: [*.go go.mod go.sum go.work go.work.sum]
[DEBUG] 16:37:22.018 properties.go:GetStringArray:52 → folders: []
[DEBUG] 16:37:22.018 properties.go:GetStringArray:52 → folders: []
[DEBUG] 16:37:22.018 properties.go:GetBool:22 → home_enabled: false
[DEBUG] 16:37:22.018 properties.go:GetBool:22 → home_enabled: false
[DEBUG] 16:37:22.018 properties.go:GetBool:22 → home_enabled: false
[DEBUG] 16:37:22.018 home.go:12 → C:\Users\Omar Izquierdo
[DEBUG] 16:37:22.018 home.go:12 → C:\Users\Omar Izquierdo
[DEBUG] 16:37:22.018 home.go:12 → C:\Users\Omar Izquierdo
[TRACE] 16:37:22.018 text.go:Render(plain) - 0s
[TRACE] 16:37:22.018 text.go:Render(plain) - 0s
[TRACE] 16:37:22.018 text.go:Render(plain) - 0s
[DEBUG] 16:37:22.018 text.go:patchTemplate:125 → in <p:blue><b>{{ .Data.Name }}</b></>
[TRACE] 16:37:22.018 text.go:Render(in <p:blue><b>{{ .Name }}</b></> ) - 0s
[TRACE] 16:37:22.018 text.go:Render(plain) - 0s
[DEBUG] 16:37:22.018 text.go:patchTemplate:125 → at <p:blue><b>{{ .Data.CurrentDate | date "15:04:05" }}</b></>
[TRACE] 16:37:22.018 text.go:Render(at <p:blue><b>{{ .CurrentDate | date "15:04:05" }}</b></>) - 0s
[TRACE] 16:37:22.018 text.go:Render(plain) - 0s
[TRACE] 16:37:22.018 terminal.go:Shell() - 0s
[TRACE] 16:37:22.018 terminal.go:Shell() - 0s
[TRACE] 16:37:22.018 terminal.go:Flags() - 0s
[TRACE] 16:37:22.018 terminal.go:Shell() - 0s
[DEBUG] 16:37:22.018 terminal_windows.go:TerminalWidth:99 → terminal width: 170
[TRACE] 16:37:22.018 terminal_windows.go:TerminalWidth() - 0s
[TRACE] 16:37:22.018 path.go:Path() - 0s
[TRACE] 16:37:22.018 terminal.go:Flags() - 0s
```

--- a/src/cli/init.go
+++ b/src/cli/init.go
@@ -77,13 +77,12 @@ func runInit(sh string) {
 	cfg, hash := config.Load(configFile, sh, false)
 
 	flags := &runtime.Flags{
-		Shell:      sh,
-		Config:     configFile,
-		Strict:     strict,
-		Debug:      debug,
-		SaveCache:  true,
-		Init:       true,
-		ConfigHash: hash,
+		Shell:     sh,
+		Config:    configFile,
+		Strict:    strict,
+		Debug:     debug,
+		SaveCache: true,
+		Init:      true,
 	}
 
 	env := &runtime.Terminal{}
@@ -97,6 +96,7 @@ func runInit(sh string) {
 	}()
 
 	feats := cfg.Features(env)
+	flags.ConfigHash = fmt.Sprintf("%s.%s", hash, feats.Hash())
 
 	var output string
 

--- a/src/segments/node.go
+++ b/src/segments/node.go
@@ -101,7 +101,7 @@ func (n *Node) loadContext() {
 		if n.env.HasFiles(pm.fileName) {
 			n.PackageManagerName = pm.name
 			n.PackageManagerIcon = n.props.GetString(pm.iconProperty, pm.defaultIcon)
-			return
+			break
 		}
 	}
 }

--- a/src/shell/features.go
+++ b/src/shell/features.go
@@ -1,5 +1,10 @@
 package shell
 
+import (
+	"strconv"
+	"strings"
+)
+
 type Feature byte
 
 const (
@@ -51,4 +56,14 @@ func (f Features) Lines(shell string) Lines {
 	}
 
 	return lines
+}
+
+func (f Features) Hash() string {
+	var sb strings.Builder
+
+	for _, feature := range f {
+		sb.WriteString(strconv.Itoa(int(feature)))
+	}
+
+	return sb.String()
 }

--- a/src/shell/filesystem.go
+++ b/src/shell/filesystem.go
@@ -57,7 +57,10 @@ func writeScript(env runtime.Environment, script string) (string, error) {
 
 	env.Cache().Set(cacheKey(env), scriptName(env), cache.INFINITE)
 
-	defer purgeScripts(env)
+	defer func() {
+		parent := filepath.Dir(path)
+		purgeScripts(env, parent)
+	}()
 
 	return path, nil
 }
@@ -128,10 +131,12 @@ func scriptPath(env runtime.Environment) (string, error) {
 	return scriptPathCache, nil
 }
 
-func purgeScripts(env runtime.Environment) {
+func purgeScripts(env runtime.Environment, path string) {
 	current := fileName(env)
-	files, err := os.ReadDir(cache.Path())
+
+	files, err := os.ReadDir(path)
 	if err != nil {
+		log.Error(err)
 		return
 	}
 
@@ -144,8 +149,11 @@ func purgeScripts(env runtime.Environment) {
 			continue
 		}
 
-		if err := os.Remove(filepath.Join(cache.Path(), file.Name())); err != nil {
+		if err := os.Remove(filepath.Join(path, file.Name())); err != nil {
 			log.Debugf("failed to remove file %s: %s", file.Name(), err)
+			continue
 		}
+
+		log.Debugf("removed old init script: %s", file.Name())
 	}
 }

--- a/src/shell/init.go
+++ b/src/shell/init.go
@@ -189,17 +189,13 @@ func sourceInit(env runtime.Environment, shell, scriptPath string, async bool) s
 
 	switch shell {
 	case PWSH, PWSH5:
-		return scriptPath
-	case ZSH:
-		return fmt.Sprintf("source '%s'", scriptPath)
-	case BASH:
-		return fmt.Sprintf("source '%s'", scriptPath)
+		return fmt.Sprintf(`& "%s"`, scriptPath)
+	case ZSH, BASH, XONSH:
+		return fmt.Sprintf(`source '%s'`, scriptPath)
 	case FISH:
-		return fmt.Sprintf("source %s", scriptPath)
+		return fmt.Sprintf(`source "%s"`, scriptPath)
 	case ELVISH:
-		return fmt.Sprintf("eval (slurp < %s)", scriptPath)
-	case XONSH:
-		return fmt.Sprintf("source %s", scriptPath)
+		return fmt.Sprintf(`eval (slurp < '%s')`, scriptPath)
 	case CMD:
 		scriptPath = strings.ReplaceAll(scriptPath, `\`, `\\`)
 		return fmt.Sprintf(`load(io.open('%s', "r"):read("*a"))()`, scriptPath)
@@ -211,13 +207,13 @@ func sourceInit(env runtime.Environment, shell, scriptPath string, async bool) s
 func sourceInitAsync(shell, scriptPath string) string {
 	switch shell {
 	case PWSH, PWSH5:
-		return fmt.Sprintf(`function prompt() { %s }`, scriptPath)
+		return fmt.Sprintf(`function prompt() { & "%s" }`, scriptPath)
 	case ZSH:
 		return fmt.Sprintf(`precmd() { source '%s' }`, scriptPath)
 	case BASH:
 		return fmt.Sprintf(`PROMPT_COMMAND='source "%s"'`, scriptPath)
 	case FISH:
-		return fmt.Sprintf(`function fish_prompt; source %s; end`, scriptPath)
+		return fmt.Sprintf(`function fish_prompt; source "%s"; end`, scriptPath)
 	default:
 		return ""
 	}

--- a/src/shell/scripts/omp.bash
+++ b/src/shell/scripts/omp.bash
@@ -143,12 +143,24 @@ function _omp_hook() {
 
 function _omp_install_hook() {
     local cmd
+    local prompt_command
+
     for cmd in "${PROMPT_COMMAND[@]}"; do
+        # skip initializing when we're already initialized
         if [[ $cmd = _omp_hook ]]; then
             return
         fi
+
+        # check if the command starts with source, if so, do not add it again
+        # this is done to avoid issues with sourcing the same file multiple times
+        if [[ $cmd = source* ]]; then
+            continue
+        fi
+
+        prompt_command+=("$cmd")
     done
-    PROMPT_COMMAND=(_omp_hook "${PROMPT_COMMAND[@]}")
+
+    PROMPT_COMMAND=(_omp_hook "${prompt_command[@]}")
 }
 
 _omp_install_hook

